octodns_cloudflare/__init__.py

```diff
@@ -1209,23 +1227,77 @@ def _apply_Delete(self, change):
                     )
                     self._try_request('DELETE', path)
 
+    def _supported_plans(self, zone_name):
+        zone_id = self.zones[zone_name]['id']
+        path = f'/zones/{zone_id}/available_plans'
+        resp = self._try_request('GET', path)
+        result = resp['result']
+        if isinstance(result, list):
+            return [plan['legacy_id'] for plan in result]
+        msg = f'{self.id}: unable to determine supported plans, do you have an Enterprise account?'
+        raise SupportsException(msg)
+
+    def _update_plan(self, zone_name, plan):
+        current_plan = self.zones[zone_name].get('plan', False)
+        if current_plan and current_plan == plan:
+            return
+        if plan in self._supported_plans(zone_name):
+            zone_id = self.zones[zone_name]['id']
+            data = {'plan': {'legacy_id': plan}}
+            resp = self._try_request('PATCH', f'/zones/{zone_id}', data=data)
+            # Update the cached plan information
+            self.zones[zone_name]['plan'] = resp['result']['plan']['legacy_id']
+        else:
+            msg = f'{self.id}: {plan} is not supported for {zone_name}'
+            raise SupportsException(msg)
+
+    def _plan_meta(self, existing, desired, changes):
+        meta = {}
```

Comment:
Would prefer to avoid looking up the plan options for every zone if the provider isn't configured to manage `plan_type`.

```suggestion
        if self.plan_type is None:
            # No plan type configured leave things unmanaged
            return
```